bigquery_etl/routine/parse_routine.py

```diff
@@ -144,186 +158,162 @@ def from_text(text, dataset, name, filepath=None, description="", is_defined=Tru
                     f"limited to chars {UDF_CHAR}, be at most 256 chars long"
                 )
 
-        # find usages of both persistent and temporary UDFs
-        dependencies = re.findall(PERSISTENT_UDF_RE, "\n".join(definitions))
-        dependencies = [".".join(t) for t in dependencies]
-        dependencies.extend(re.findall(TEMP_UDF_RE, "\n".join(definitions)))
-
-        # for public UDFs dependencies can live in arbitrary dataset;
-        # we can check if some known dependency is part of the UDF
-        # definition instead
-        for udf in MOZFUN_ROUTINES:
+        # get routines that could be referenced by the UDF
+        routines = get_routines(project)
+        dependencies = []
+        for udf in routines:
             if udf["name"] in "\n".join(definitions):
                 dependencies.append(udf["name"])
 
+        dependencies.extend(re.findall(TEMP_UDF_RE, "\n".join(definitions)))
+        dependencies = list(set(dependencies))
```

Comment:
I don't think deduping is necessary anymore, since we're iterating through the list of routines (which should have no duplicates in itself).

Reply:
It's still necessary for when routines in `mozfun` are parsed. `mozfun` routines are by default part of `get_routines` since they can be referenced by any routine in any project. So if `mozfun` is passed in as project routines should be parsed from then these UDFs will appear twice.